src/components/sync-configurator/BaseSyncConfigurator.jsx

```diff
@@ -95,30 +138,75 @@ class BaseSyncConfigurator extends React.Component {
         }
     };
 
-    handleDialogClose = importResponse => {
-        if (importResponse) {
-            this.setState({ syncDialogOpen: false, syncSummaryOpen: true, importResponse });
-        } else {
-            this.props.snackbar.error(i18n.t("Unknown error with the request"));
-            this.setState({ syncDialogOpen: false });
-        }
+    getGroupFilterData = async () => {
+        const { d2, groupFilterName } = this.props;
+
+        const groupClass = d2ModelFactory(d2, groupFilterName);
```

Comment:
`props.groupFilterName` is optional. I know the condition has been tested by the caller, but a method should work on isolation. So one option is checking here the prop instead of the parent, and return a null (so the parent is simplified). Idem for getLevelFilterData.